tools/smoother_benchmarking/README.md

```diff
@@ -0,0 +1,121 @@
+# Planners Smoothing Benchmark
+
+This experiment runs a set with randomly generated goals for objective benchmarking.
+
+Bechmarking scripts require the following python packages to be installed:
+
+```
+pip install transforms3d
+pip install seaborn
+pip install tabulate
+```
+
+To perform the benchmarking, the following changes should be made for planner and smoother servers:
+
+```
+diff --git a/nav2_planner/src/planner_server.cpp b/nav2_planner/src/planner_server.cpp
+index c7a90bcb..510bb6ea 100644
+--- a/nav2_planner/src/planner_server.cpp
++++ b/nav2_planner/src/planner_server.cpp
+@@ -285,7 +285,7 @@ bool PlannerServer::getStartPose(
+   typename std::shared_ptr<const typename T::Goal> goal,
+   geometry_msgs::msg::PoseStamped & start)
+ {
+-  if (goal->use_start) {
++  if (true) {
+     start = goal->start;
+   } else if (!costmap_ros_->getRobotPose(start)) {
+     return false;
+@@ -381,7 +381,10 @@ void PlannerServer::computePlanThroughPoses()
+       }
+ 
+       // Get plan from start -> goal
++      auto planning_start = steady_clock_.now();
+       nav_msgs::msg::Path curr_path = getPlan(curr_start, curr_goal, goal->planner_id);
++      auto planning_duration = steady_clock_.now() - planning_start;
++      result->planning_time = planning_duration;
```

Comment:
Why move this?

Reply:
Initially, the time measured for planner and smoother actions - is a whole period of time required for the selected action. This time includes a lot of of supplementary cycles, like TF to global frame or waiting for costmaps. During the benchmarking, we need to know exact time of `getPlan()` for the selected path planner and `smooth()` routine for smoothers case. That is why for the most accurate results, start and end time measurements should be moved right before and after measured routines, to exclude all unnecessary wrapping/transformation/converting/preparation/etc. stuff